extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/KubernetesOverlordModuleTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord;
+
+import com.google.inject.Provider;
+import org.apache.druid.indexing.overlord.RemoteTaskRunnerFactory;
+import org.apache.druid.indexing.overlord.hrtr.HttpRemoteTaskRunnerFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KubernetesOverlordModuleTest
+{
+
+  @Mock
+  private Provider<RemoteTaskRunnerFactory> parentProvider;
+  @Mock
+  private KubernetesAndWorkerTaskRunnerConfig runnerConfig;
+  @Mock
+  private RemoteTaskRunnerFactory remoteTaskRunnerFactory;
+
+  private KubernetesOverlordModule module;
+
+  @Before
+  public void setUp()
+  {
+    module = new KubernetesOverlordModule();
+  }
+
+  @Test
+  public void test_provideNullRemoteTaskRunner_withHttpRunnerType()
+  {
+    when(runnerConfig.getWorkerType()).thenReturn(HttpRemoteTaskRunnerFactory.TYPE_NAME);
+    assertNull(module.provideRemoteTaskRunnerFactory(parentProvider, runnerConfig));
+  }
+
+  @Test
+  public void test_provideRemoteTaskRunner_withRemoteRunnerType()
+  {
+    when(runnerConfig.getWorkerType()).thenReturn(RemoteTaskRunnerFactory.TYPE_NAME);
+    when(parentProvider.get()).thenReturn(remoteTaskRunnerFactory);
+    assertSame(remoteTaskRunnerFactory, module.provideRemoteTaskRunnerFactory(parentProvider, runnerConfig));
+  }
```

Comment:
Can we add a test here to see that the KubernetesAndWorkerTaskRunnerFactory is initialized correctly please

Reply:
I updated the code, this Module change is reverted.